mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBDatastoreProvider.java

```diff
@@ -168,4 +168,20 @@ private DB extractDatabase(MongoClient mongo, MongoDBConfiguration config) {
 			}
 		}
 	}
+
+	protected MongoDBConfiguration getConfig() {
+		return config;
+	}
+
+	protected MongoClient getMongoClient() {
+		return mongo;
+	}
+
+	protected void setMongoClient(MongoClient mongoClient) {
```

Comment:
Maybe you can get rid of these when pursuing the delegation approach and pass in stuff via the constructor?